patches/api/0463-Expand-PlayerDeathEvent-API.patch

```diff
@@ -0,0 +1,118 @@
+         this.newLevel = newLevel;
+         this.deathMessage = deathMessage;
+         this.doExpDrop = doExpDrop; // Paper - shouldDropExperience API
++        this.deathScreenMessage = deathMessage; // Paper - Expand PlayerDeathEvent API
++        this.displayDeathMessage = displayDeathMessage; // Paper - Expand PlayerDeathEvent API
+     }
+     // Paper end - adventure
+ 
++    @Deprecated // Paper
+     public PlayerDeathEvent(@NotNull final Player player, @NotNull final List<ItemStack> drops, final int droppedExp, @Nullable final String deathMessage) {
+         this(player, drops, droppedExp, 0, deathMessage);
+     }
+@@ -69,6 +76,8 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         this.newLevel = newLevel;
+         this.deathMessage = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserializeOrNull(deathMessage); // Paper
+         this.doExpDrop = doExpDrop; // Paper - shouldDropExperience API
++        this.deathScreenMessage = this.deathMessage; // Paper - Expand PlayerDeathEvent API
++        this.displayDeathMessage = true; // Paper - Expand PlayerDeathEvent API
+     }
+ 
+     @Deprecated // Paper
+@@ -160,6 +169,49 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         return this.deathMessage;
+     }
+     // Paper end - adventure
++    // Paper start - Expand PlayerDeathEvent API
++    /**
++     * Set the death message that will appear on the death screen of the dying player.
++     * <p>
++     * If set to null, no message will be shown to the dying player.
++     * <p>
++     * If the message exceeds 256 characters it will be truncated.
++     *
++     * @param deathScreenMessage Message to appear on the death screen to the dying player.
++     */
++    public void deathScreenMessage(@Nullable net.kyori.adventure.text.Component deathScreenMessage) {
++        this.deathScreenMessage = deathScreenMessage;
++    }
++
++    /**
++     * Get the death message that will appear on the death screen of the dying player.
++     * By default, it is the same value as {@link #deathMessage()}.
++     *
++     * @return Message to appear on the death screen to the dying player.
++     */
++    public @Nullable net.kyori.adventure.text.Component deathScreenMessage() {
++        return this.deathScreenMessage;
++    }
++
++    /**
++     * Returns whether the death message in this event should be displayed according to
++     * `showDeathMessages` game rule.
++     *
++     * @return whether the death message in this event should be displayed
++     */
++    public boolean shouldDisplayDeathMessage() {
++        return displayDeathMessage;
++    }
++
++    /**
++     * Sets whether the death message in this event should be shown.
++     *
++     * @param displayDeathMessage whether the death message in this event should be shown.
++     */
++    public void setDisplayDeathMessage(boolean displayDeathMessage) {
```

Comment:
This name is kind of misleading. `setDisplayDeathMessage` sounds like it sets the message, when in reality it only sets if the message is shown.

Reply:
Would setShouldDisplayDeathMessage be better? :D Or should I also omit the get/set prefixes?